File: week01/alanhuang/342_PowerOfFour.java
/**
 * @Classname PowerOfFour
 * @Description TODO
 * @Date 2020/9/7 10:15
 * @Author guangliang huang
 * @Copyright:2020 linkcm.com Inc. All rights reserved.
 * 注意：本内容仅限于公司内部传阅，禁止外泄以及用于其他的商业目的
 */
public class PowerOfFour {
	public boolean isPowerOfFour(int num) {
		if(num == 0){
			return false;
		}
		while(num % 4 == 0){
			num /= 4;
		}
		return num == 1;
	}
}

File: week01/siu/342_power_of_four.go
package siu

func isPowerOfFour(num int) bool {
	// 4的幂：正整数
	// 边界：0，1~3，4，5~7
	// 0: flase；与1对比
	// 1~3: x / 4 = 0：取余
	// 4: x / 4 = 1；与1对比
	// 5~7: x / 4 = 1；取余
	for num != 0 && num%4 == 0 {
		num = num / 4
	}
	return num == 1
}
